pymodbus/client/base.py

```diff
@@ -108,6 +108,10 @@ async def async_execute(self, request) -> ModbusResponse:
             async with self._lock:
                 req = self.build_response(request)
                 self.ctx.send(packet)
+                no_response_expected = hasattr(request, "no_response_expected") and request.no_response_expected is True
```

Comment:
hasattr is relative costly, please add no_response_expected to ModbusPDU so that it is always available.

Reply:
Certainly